model/project.go

```diff
@@ -941,12 +941,30 @@ func (p *Project) BuildProjectTVPairsWithAlias(alias string) ([]TVPair, error) {
 		for _, variant := range p.BuildVariants {
 			if variantRegex.MatchString(variant.Name) {
 				for _, task := range p.Tasks {
-					if ((v.Task != "" && taskRegex.MatchString(task.Name)) ||
-						(len(v.Tags) > 0 && len(util.StringSliceIntersection(task.Tags, v.Tags)) > 0)) &&
-						(p.FindTaskForVariant(task.Name, variant.Name) != nil) {
+					if task.Patchable != nil && !(*task.Patchable) {
+						continue
+					}
+					if !((v.Task != "" && taskRegex.MatchString(task.Name)) ||
+						(len(v.Tags) > 0 && len(util.StringSliceIntersection(task.Tags, v.Tags)) > 0)) {
+						continue
+					}
+
+					if p.FindTaskForVariant(task.Name, variant.Name) != nil {
 						pairs = append(pairs, TVPair{variant.Name, task.Name})
 					}
 				}
+
+				for _, displayTask := range variant.DisplayTasks {
+					if v.Task == "" || !taskRegex.MatchString(displayTask.Name) {
+						continue
+					}
+
+					for _, task := range displayTask.ExecutionTasks {
```

Comment:
You shouldn't need this block - just add the display task name and let extractDisplayTasks handle the rest

Reply:
I think the UI depends on having the task list in the build variant (for notional performance reasons,)

Maybe it makes sense to push more data up into the build variants?

Reply:
I was referring deleting the tasks list in the Patch, which is also the argument to extractDisplayTasks, and also what I thought I thought John was referring to. @john-m-liu were you referring to this tasks list or the one in patch? Regardless, I think this loop is still needed.

Reply:
I was mistaken about which parameter of extractDisplayTasks that this ends up being funneled in to, but I think the approach here is still not quite right. The display task that is matched needs to end up in the DisplayTasks property of the struct returned by extractDisplayTasks, which it won't with the current logic

Reply:
Got it, addressed in latest push